### PR TITLE
changed translation function with vectorization

### PR DESCRIPTION
**Pull request by lingminhao**
I changed the translation function to using the apply families (vectorization). I still have to filter the stop codons though.

### CODE REVIEW

File: R/translation.R [hunk 1]
```diff
@@ -8,15 +8,16 @@
 #' @examples
 #' rna <- "AUGGCCAUGGCGCCCAGAACUGAGAUCAAUAGUACCCGUAUUAACGGGUGA"
 #' translation(rna)
+
 translation <- function(rna){
-  amino <- ""
-  for(i in seq(1, nchar(rna), by = 3)){
-    if (codon_table[substr(rna,i,i+2),] == "Stop"){
+  rna_split <- substring(rna, seq(1, nchar(rna),3), seq(3, nchar(rna),3))
+  for (i in seq(length(rna_split))){
```
Comment (cying111) [inline R/translation.R:14]:
```suggestion
  for (i in seq_along(rna_split)){
```

File: R/translation.R [hunk 2]
```diff
@@ -8,15 +8,16 @@
...
+    if (codon_table[rna_split[i],] == "Stop"){
+      rna_split <- rna_split[1:i-1]
```
Comment (cying111) [inline R/translation.R:16]:
```suggestion
      rna_split <- rna_split[1:(i-1)]
```

Comment (cying111) [inline R/translation.R:16]: I know it's probably not possible, but just in case, if maximum of length rna_split is 1, this might break?

Comment (lingminhao) [reply R/translation.R:16]: I think it will give empty string, but I will add a unit test to verify it.

**Review by lingminhao — COMMENTED** [review]
.

File: tests/testthat/test-translation.R [hunk 3]
```diff
@@ -4,4 +4,5 @@ test_that("translation() converts RNA to Amino acid", {
 
 test_that("translation() terminates immediately for stop codon", {
   expect_equal(translation("AUGUGAAUGGCGCCCAGAACUGAGAUCAAUAGUACCCGUAUUAACGGGUGA"), "M")
+  expect_equal(translation("UGA"), "")
```
Comment (lingminhao) [inline tests/testthat/test-translation.R:7]: It passed the unit test and gave an empty string for rna_split with max length 1.

**Pull request merged**